src/client/app/utils/calibration.ts

```diff
@@ -86,25 +122,82 @@ export function isValidGPSInput(input: string) {
-	const mapScale: MapScale = calculateScale(originComplete, oppositeComplete);
-	return mapScale;
+/**
+ * Calculates the GPS unit per coordinate unit.
+ * @param origin The GPS value for the origin that was computed during calibration.
+ * 	This is the bottom, left corner of the user map.
+ * @param opposite The GPS value for the opposite that was computed during calibration.
+ * 	This is the top, right corner of the user map.
+ * @param imageDimensions This is the size of the normalized map image.
+ * @returns a pair {deltaX, deltaY} representing the GPS degree per unit (x, y)
+ * 	on true north map grid.
+ */
+export function calculateScaleFromEndpoints(origin: GPSPoint, opposite: GPSPoint, size: Dimensions): MapScale {
+	// GPS values do not need to shift or rotate. The origin/opposite needs to be on the true north map.
+	// See calibrate function for the cartesian values used.
+	const originComplete: CalibratedPoint = { gps: origin, cartesian: trueNorthOrigin(size) };
+	const oppositeComplete: CalibratedPoint = { gps: opposite, cartesian: trueNorthOpposite(size) };
+	const scaleOfMap: MapScale = calculateScale(originComplete, oppositeComplete);
+	return scaleOfMap;
 }
 
-export function calibrate(calibrationSet: CalibratedPoint[], imageDimensions: Dimensions) {
+/**
+ * Convert the gps value to the equivalent Plotly grid coordinates on user map.
+ * @param size The normalized size of the map
+ * @param gps The GPS coordinate to convert
+ * @param originGPS The GPS value of the origin on the true north map.
+ * @param scaleOfMap The GPS degree per unit x, y on the true north map.
+ * @returns x, y value of the gps point on the user map.
+ */
+export function gpsToUserGrid(size: Dimensions, gps: GPSPoint, originGPS: GPSPoint, scaleOfMap: MapScale): CartesianPoint {
+	// We need the origin x, y value by starting from 0, 0 on the user map and
+	// shift/rotate into true north.
+	const originTrueNorth = trueNorthOrigin(size);
+	// Now, convert from GPS to true north grid (x, y).
+	// Calculate how far the point is from origin and then the units for this distance.
+	const gridTrueNorth: CartesianPoint = {
+		x: originTrueNorth.x + (gps.longitude - originGPS.longitude) / scaleOfMap.degreePerUnitX,
+		y: originTrueNorth.y + (gps.latitude - originGPS.latitude) / scaleOfMap.degreePerUnitY
+	};
+	// Now rotate to user map and shift so origin at bottom, left. Do - angle since going from
+	// true north to user map.
+	const userGrid: CartesianPoint = rotateShift(size, gridTrueNorth, 1, -trueNorthAngle.angle);
+	return userGrid;
+}
+
+/**
+ * @param calibrationSet All the points clicked by the user for calibration.
+ * @param imageDimensions The dimensions of the original map to use from the user.
+ * @returns The error and the origin & opposite point in GPS to use for mapping.
+ */
+export function calibrate(calibrationSet: CalibratedPoint[], imageDimensions: Dimensions): CalibrationResult {
```

Comment:
I suppose I'm being nitpicky here, but I think there's supposed to be a short summary before the params like for the other functions.